src/OidcClient.php

```diff
@@ -130,6 +136,21 @@ public function generateAuthorizationRedirect(
       $data['prompt'] = $prompt;
     }
 
+    try {
+      // Check if code challenge is activated
+      if (null !== $this->codeChallengeMethod) {
+        $codeVerifier  = $this->generateCodeVerifier();
+        $codeChallenge = $this->generateCodeChallenge($codeVerifier);
+
+        $data = array_merge($data, [
+            'code_challenge'        => $codeChallenge,
```

Comment:
```suggestion
            'code_challenge'        => $this->generateCodeChallenge(),
```

I believe the `generateCodeVerifier()` and `generateCodeChallenge()` can be collapsed in a single method, right?

Reply:
It can be, although only the code verifier needs to be in the session in order to pass it along in the `requestTokens` call. I will have a look at it.

Reply:
Merged them together